WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -669,6 +683,78 @@ private extension EditableOrderViewModel {
+        orderSynchronizer.setProduct.send(input)
+
+        analytics.track(event: WooAnalyticsEvent.Orders.orderProductAdd(flow: flow.analyticsFlow))
+    }
+
+    /// Removes a selected product (from the product list) to the order.
+    ///
+    func removeProductFromOrder(_ product: Product) {
+        guard let orderItem = orderSynchronizer.order.items.first(where: { $0.productID == product.productID }) else {
+            DDLogError("Unable to find product ID: \(product.productID) in Order")
+            return
+        }
+        removeItemFromOrder(orderItem)
+        analytics.track(event: WooAnalyticsEvent.Orders.orderProductRemove(flow: flow.analyticsFlow))
+    }
+
+    /// Removes a selected product variation (from the product variation list) to the order.
+    ///
+    func removeProductVariationFromOrder(_ variation: ProductVariation) {
+        guard let orderItem = orderSynchronizer.order.items.first(where: { $0.variationID == variation.productVariationID }) else {
+            DDLogError("Unable to find product variation ID: \(variation.productVariationID) (parent ID: \(variation.productID)) in Order")
+            return
+        }
+        removeItemFromOrder(orderItem)
+        analytics.track(event: WooAnalyticsEvent.Orders.orderProductRemove(flow: flow.analyticsFlow))
+    }
+
+    /// Handles a product by adding it to the Order if selected, or removing it from the Order if unselected
+    ///
+    func handleProduct(_ product: Product) {
+        // All products
+        if !allProducts.contains(product) {
+            allProducts.append(product)
+        }
+
+        if !selectedProducts.contains(product) {
+            // Add Product to the Order
+            selectedProducts.append(product)
+            addProductToOrder(product)
+        } else {
+            // Remove Product from the Order
+            selectedProducts.removeAll(where: { $0?.productID == product.productID })
+            removeProductFromOrder(product)
+        }
+    }
+
+    /// Handles a product variation by adding it to the Order if selected, or removing it from the Order if unselected
+    ///
+    func handleProductVariation(_ variation: ProductVariation, parent product: Product) {
+        if !allProducts.contains(product) {
+            allProducts.append(product)
+        }
+
+        if !allProductVariations.contains(variation) {
+            allProductVariations.append(variation)
+        }
+
+        if !selectedProductVariations.contains(variation) {
+            // Add Product variation to the Order
+            selectedProductVariations.append(variation)
+            addProductVariationToOrder(variation)
+        } else {
+            // Remove Product variation from the Order
+            selectedProductVariations.removeAll(where: { $0?.productVariationID == variation.productVariationID })
+            removeProductVariationFromOrder(variation)
+        }
+    }
```

Comment:
As there's certain logic repetition, would you say is worth it to attempt and make an unique method here for both `handleProduct` and `handleProductVariation`? Or to keep these separate for readability?